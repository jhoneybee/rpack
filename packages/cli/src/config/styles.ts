import * as Config from 'webpack-chain';

export const presetCss = (config: Config) => {
    config
        .module
        .rule('css')
        .test(/\.css$/)
        .use('style-loader')
        .loader('style-loader')
        .end()
        .use('css/loader')
        .loader('css-loader')
        .end();
};

export const presetLess = (config: Config) => {
    config
        .module
        .rule('less')
        .test(/\.less$/)
        .use('style-loader')
        .loader('style-loader')
        .end()
        .use('css/loader')
        .loader('css-loader')
        .end()
        .use('less/loader')
        .loader('less-loader')
        .options({
            lessOptions: {
                javascriptEnabled: true,
            },
        })
        .end();
};
