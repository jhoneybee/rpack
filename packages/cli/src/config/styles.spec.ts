import * as Config from 'webpack-chain';
import { presetCss, presetLess } from './styles';

test('test presetCss', () => {
    const config = new Config();
    presetCss(config);
    expect(config.toConfig()).toMatchSnapshot();
});

test('test presetLess', () => {
    const config = new Config();
    presetLess(config);
    expect(config.toConfig()).toMatchSnapshot();
});
