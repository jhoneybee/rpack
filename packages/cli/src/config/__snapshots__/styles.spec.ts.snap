// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test presetCss 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "style-loader",
          },
          Object {
            "loader": "css-loader",
          },
        ],
      },
    ],
  },
}
`;

exports[`test presetLess 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.less\\$/,
        "use": Array [
          Object {
            "loader": "style-loader",
          },
          Object {
            "loader": "css-loader",
          },
          Object {
            "loader": "less-loader",
            "options": Object {
              "lessOptions": Object {
                "javascriptEnabled": true,
              },
            },
          },
        ],
      },
    ],
  },
}
`;
